### PR TITLE
Eq 847 remove jquery

### PR DESCRIPTION
**Pull request by bennpearson**
### What is the context of this PR?
Removed jQuery because it is only used for one pattern so have replaced this with javascript and used polyfills.js to add support to older browsers.  

### How to review 
Check focus on radio and checkbox inputs.
Also check browsers that don't support the following methods:
- https://developer.mozilla.org/en-US/docs/Web/API/Element/closest
- https://developer.mozilla.org/en/docs/Web/API/Element/classList

### CODE REVIEW

File: app/assets/js/app/modules/focus-styles.js [hunk 1]
```diff
@@ -7,23 +7,49 @@ const checkedClass = 'is-checked'
...
-        // uncheck siblings
-        $closestFocusableBox.siblings(`.${focusableBoxClass}`).removeClass(checkedClass)
+  const checkedElements = Array.from(document.getElementsByClassName(focusableClass))
+  for (let checkedElement of checkedElements) {
+    if (checkedElement.checked === true) {
+      checkedElement.closest(`.${focusableBoxClass}`).classList.add(checkedClass)
+    }
+  }
+  function getNextSiblings(e, filter) {
+    var siblings = []
```
Comment (hamishtaplin) [inline app/assets/js/app/modules/focus-styles.js]: don't use `var`—use `let` or `const` where appropriate

might be worth adding a linting rule for this too: http://eslint.org/docs/rules/no-var

File: app/assets/js/app/modules/focus-styles.js [hunk 2]
```diff
@@ -7,23 +7,49 @@ const checkedClass = 'is-checked'
...
-      if ($this.attr('type') === 'radio') {
-        // uncheck siblings
-        $closestFocusableBox.siblings(`.${focusableBoxClass}`).removeClass(checkedClass)
+  const checkedElements = Array.from(document.getElementsByClassName(focusableClass))
+  for (let checkedElement of checkedElements) {
+    if (checkedElement.checked === true) {
+      checkedElement.closest(`.${focusableBoxClass}`).classList.add(checkedClass)
+    }
+  }
+  function getNextSiblings(e, filter) {
```
Comment (hamishtaplin) [inline app/assets/js/app/modules/focus-styles.js]: move these into a helper called selectors and then `import`

**Review by hamishtaplin — CHANGES_REQUESTED** [review]

**Comment by hamishtaplin** [comment]
Still need to add [linting rule](http://eslint.org/docs/rules/no-var) to discourage use of `var` through a warning.

**Review by hamishtaplin — APPROVED** [review]

File: .eslintrc [hunk 3]
```diff
@@ -26,6 +26,7 @@
     "consistent-return": 0,
     "no-underscore-dangle": 0,
     "indent": [2, 2, {"SwitchCase": 1}],
-    "space-before-blocks": [2, "always"]
+    "space-before-blocks": [2, "always"],
+    "no-var": [1]
```
Comment (benjefferies) [inline .eslintrc:30]: Is this something we want to turn off globally or just ignore on the case by case scenario?

**Review by benjefferies — CHANGES_REQUESTED** [review]

File: .eslintrc (hunk 3 above)
Comment (hamishtaplin) [reply .eslintrc:30]: @benjefferies There's no reason to use `var` unless you specifically need to take advantage of the way it's scoped—which is highly unlikely.

**Review by benjefferies — APPROVED** [review]

File: .eslintrc (hunk 3 above)
Comment (benjefferies) [reply .eslintrc:30]: Though this was disabling it as I thought it was enabled by default

**Pull request merged**